docs/i18n-docs.md

```diff
@@ -0,0 +1,189 @@
+---
+title: Automation tools for i18n and translate
+authors:
+- "@chouchongYHMing" # Authors' github accounts here.
+reviewers:
+- TBD
+approvers:
+- "@warjiang"
+
+creation-date: 2024-08-19
+
+---
+
+# Automation tools for i18n and translate
+
+## Summary
+
+This is an out-of-the-box automated i18n tool designed for the Karmada dashboard, developed primarily from a Chinese environment perspective. It implements the following features:
+
+1. Automatically configures i18n key-value pairs in the form of {key:value}.
+2. Automatically translates all i18n key-value pairs in a directory and adds them to the existing i18n repository.
+
+<!--
+这是一个为Karmada dashboard设计的开箱即用的自动化i18n工具，基于以中文环境开发的主视角，它实现了以下功能：
+1. 自动配置i18n键值对，并且为{key:value}形式。
+2. 自动翻译整个目录下的i18n键值对，并且添加到原有的i18n仓库中。
+-->
+
+## Motivation
+
+The global digital landscape demands applications that cater seamlessly to diverse linguistic audiences. Recognizing this need, we embarked on creating an automated tool designed specifically for the Karmada dashboard, which simplifies the process of internationalizing applications by automatically detecting and converting Chinese text within the code into internationalized function calls. This tool not only enhances the accessibility and usability of applications across different language-speaking regions but also streamlines the development process by integrating internationalization early in the software development lifecycle. 
+<!--
+全球数字环境要求应用程序能够无缝地迎合不同语言的受众。认识到这一需求，我们着手创建一个专门为karma仪表板设计的自动化工具，通过自动检测并将代码中的中文文本转换为国际化的函数调用，简化了国际化应用程序的过程。该工具不仅增强了跨不同语言区域的应用程序的可访问性和可用性，而且还通过在软件开发生命周期的早期集成国际化来简化开发过程。通过自动生成和更新用于国际化的键值对。
+-->
+
+### Goals
+
+This tool aims to automatically identify various code nodes, like plain strings and JSX elements, replacing Chinese characters with i18n keys and integrating them into locales/zh-CN.json. It then executes translations based on this file to produce multilingual content in languages such as English.
+
+<!--
+该工具旨在自动识别各种代码节点，如普通字符串和JSX元素，用i18n键替换中文字符并将其集成到locale /zh-CN.json中。然后，它基于该文件执行翻译，以生成英语等语言的多语言内容。
+-->
+
+### Non-Goals
+
+Developers need to manually add the styles of Chinese characters types that are not already supported by the tool when encountering them.
+
+<!--
+开发人员需要在遇到工具不支持的汉字样式时手动添加样式。
+-->
+
+## Proposal
+
+1. Automatic Node Identification: Enhance the tool to automatically identify and process various node types within the source code, including plain string nodes, template literal nodes, JSX text nodes, and JSX attribute nodes. This capability will ensure comprehensive scanning and modification of all potential text elements that may require internationalization.
```

Comment:
this section seems very similar as the Design Details, maybe we can simplify these two sections